core/lib/jobs/jobs/jobs.py

```diff
@@ -213,3 +217,30 @@ def _get_placeholders_str(num_placeholders, start=1):
     if start < 1:
         raise ValueError("Initial placeholder value must be >= 1")
     return ", ".join(f"${i}" for i in range(start, start + num_placeholders))
+
+
+async def check_customer_quota(con, customer_id, service) -> Dict[str, bool]:
+    """Query DB for service-specific customer account usage.
+
+    Will be called for all account tiers, unlimited has a value of -1.
+    Returns:
+        - Dictionary containing boolean values for if uploads or job quotas have been reached
+    """
+
+    async with con.transaction():
+        # get service specific usage restrictions for the customer account
+        usage_query = "SELECT usage_restrictions->$1 AS usage FROM customers WHERE id=$2"
+        usage_json = await con.fetchrow(usage_query, service, customer_id)
+        usage_dict = json.loads(usage_json["usage"])
+
+        # grab total uploads and jobs of customer for specific service
+        query = "SELECT COUNT(*) as total_uploads, SUM(jobs_count) as total_jobs FROM (SELECT COUNT(*) AS jobs_count FROM jobs_result WHERE customer_id=$1 AND type=$2 GROUP BY upload_id) dt"
+        customer_data = await con.fetchrow(query, customer_id, service)
+        total_uploads = customer_data["total_uploads"]
+        total_jobs = customer_data["total_jobs"] if customer_data["total_jobs"] is not None else 0
```

Comment:
Would the `total_jobs` value returned by the query ever be `None`? I'm honestly not sure

Reply:
New users get None returned from the SUM if there is nothing to sum